internal/services/mssql/mssql_job_step_resource.go

```diff
@@ -0,0 +1,445 @@
+
+			existing, err := client.Get(ctx, id)
+			if err != nil && !response.WasNotFound(existing.HttpResponse) {
+				return fmt.Errorf("checking for presence of existing %s: %+v", id, err)
+			}
+
+			if !response.WasNotFound(existing.HttpResponse) {
+				return metadata.ResourceRequiresImport(r.ResourceType(), id)
+			}
+
+			parameters := jobsteps.JobStep{
+				Name: pointer.To(model.Name),
+				Properties: pointer.To(jobsteps.JobStepProperties{
+					Action: jobsteps.JobStepAction{
+						Value: model.SqlScript,
+					},
+					Credential: pointer.To(model.JobCredentialID),
+					ExecutionOptions: pointer.To(jobsteps.JobStepExecutionOptions{
+						InitialRetryIntervalSeconds:    pointer.To(model.InitialRetryIntervalSeconds),
+						MaximumRetryIntervalSeconds:    pointer.To(model.MaximumRetryIntervalSeconds),
+						RetryAttempts:                  pointer.To(model.RetryAttempts),
+						RetryIntervalBackoffMultiplier: pointer.To(model.RetryIntervalBackoffMultiplier),
+						TimeoutSeconds:                 pointer.To(model.TimeoutSeconds),
+					}),
+					StepId:      pointer.To(model.JobStepIndex),
+					TargetGroup: model.JobTargetGroupID,
+				}),
+			}
+
+			if len(model.OutputOptions) != 0 {
+				outputOptions := model.OutputOptions[0]
+
+				databaseId, err := commonids.ParseSqlDatabaseID(outputOptions.MsSqlDatabaseId)
+				if err != nil {
+					return err
+				}
+
+				parameters.Properties.Output = pointer.To(jobsteps.JobStepOutput{
+					Credential:        pointer.To(outputOptions.JobCredentialId),
+					DatabaseName:      databaseId.DatabaseName,
+					ResourceGroupName: pointer.To(databaseId.ResourceGroupName),
+					SchemaName:        pointer.To(outputOptions.SchemaName),
+					ServerName:        databaseId.ServerName,
+					SubscriptionId:    pointer.To(databaseId.SubscriptionId),
+					TableName:         outputOptions.TableName,
+				})
+			}
+
+			if _, err := client.CreateOrUpdate(ctx, id, parameters); err != nil {
+				return fmt.Errorf("creating %s: %+v", id, err)
+			}
+
+			metadata.SetID(id)
+			return nil
+		},
+	}
+}
+
+func (r MsSqlJobStepResource) Read() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 5 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			client := metadata.Client.MSSQL.JobStepsClient
+
+			id, err := jobsteps.ParseStepID(metadata.ResourceData.Id())
+			if err != nil {
+				return err
+			}
+
+			resp, err := client.Get(ctx, *id)
+			if err != nil {
+				if response.WasNotFound(resp.HttpResponse) {
+					return metadata.MarkAsGone(id)
+				}
+
+				return fmt.Errorf("retrieving %s: %+v", id, err)
+			}
+
+			state := MsSqlJobStepResourceModel{
+				Name:  id.StepName,
+				JobID: jobsteps.NewJobID(id.SubscriptionId, id.ResourceGroupName, id.ServerName, id.JobAgentName, id.JobName).ID(),
+			}
+
+			if model := resp.Model; model != nil {
+				if props := model.Properties; props != nil {
+					state.JobCredentialID = pointer.From(props.Credential)
```

Comment:
Since casing can be so variable in Azure, we have a practice of parsing any resource IDs returned by the API with our ID parsers to:
1. Ensure the API is returning the expected ID type
2. To validate the casing of the static ID segments and to normalise them if required

This prevents any spurious diffs caused by casing variations.

My approach is to parse the ID using the regular parser (not the insensitive one), this allows us to catch any change in behaviour or bugs in the API if it begins returning an ID with the wrong casing. If/when this happens we can then switch to using the case insensitive parser to workaround this behaviour and normalise the casing.

```suggestion
					if v := pointer.From(props.Credential);  != "" {
					       credentialID, err := jobcredentials.ParseCredentialID(v)
					       if err != nil {
					              return err
					       }
					       state.JobCredentialID = credentialID.ID()
					}
```